extension/content.js

```diff
@@ -108,16 +108,43 @@ function getVideoId() {
     }
 }
 
-function fetchComments(videoId) {
-    const part = 'snippet'
-    const fields = 'items(snippet(topLevelComment(snippet)))'
-    const order = 'relevance'
-    return fetch(`https://www.googleapis.com/youtube/v3/commentThreads?videoId=${videoId}&part=${part}&fields=${fields}&order=${order}&key=${API_KEY}`)
+function fetchAllComments(videoId) {
+    return new Promise(async (resolve) => {
+        let items = [];
+        let numberOfPagesToFetch = 5;
```

Comment:
```suggestion
        let numberOfPagesToFetch = 3
```

I would limit it to 3 pages (300 comments in case of 100 comments per page) or event 2:
- Every comment spends [quota](https://developers.google.com/youtube/v3/getting-started#quota).
- Standard YT page is 20 comments, so it's like 15 pages of scrolling for a user in our case.